### PR TITLE
[CI] Update Unit Test/Lint Workflows

### PR DESCRIPTION
**Pull request by langong347**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #334
* __->__ #326
* #322
* #323
* #333

Summary:

- Update github action versions
- Remove conda activate as conda action will create *and activate* the environment
- Use `env`, `defaults` to avoid repetitive usage of variables, commands
- Prepend python commands with `python -m` to ensure we get the correct version
  - Except for `pytest`, reason see #322
- Add a step to upgrade `pip`
- Use default fetch depth `1` when checking out the repo; `0` fetches all history/branches/tags.
- Add test profiling

Test Plan:

- CI

Differential Revision: [D39904702](https://our.internmc.facebook.com/intern/diff/D39904702)

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
-      - name: Setup Python
-        uses: actions/setup-python@v2
-        with:
-          python-version: "3.8"
-      - name: Check out repos
-        uses: actions/checkout@v2
+      - name: Check out repo
+        uses: actions/checkout@v3
+      - name: Setup python
+        uses: actions/setup-python@v4
```
[the remark this answers is not in the document]
Comment (langong347) [reply .github/workflows/lint.yaml:20]: Github Action MarketPlace (unfortunately it's not automatic to update): https://github.com/marketplace/actions/setup-python

@pikapecan

**Review by pikapecan — APPROVED** [review]

**Comment by langong347** [comment]
@langong347 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39904702).

**Comment by langong347** [comment]
@langong347 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39904702).

**Comment by langong347** [comment]
@langong347 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39904702).

**Pull request closed without merging**